Refresh documentation about CPU & RAM limits; (#1289)
codenvy/che-docs: src/main/pages/che-7/administration-guide/proc_workspaces-requirements.adoc

+
... Custom memory limit and request:
+

> If present, the `memoryLimit` and `memoryRequest` attributes of the `containers` section of the `meta.yaml` file defines the memory limit of a `chePlugins` or `cheEditors` component. {prod-short} automatically sets the memory request to match the memory limit in case it is not specified explicitly.

Reviewer: ```suggestion
If present, the `memoryLimit` and `memoryRequest` attributes of the `containers` section of the `meta.yaml` file define the memory limit of the `chePlugins` or `cheEditors` components. {prod-short} automatically sets the memory request to match the memory limit in case it is not specified explicitly.
```

... Custom CPU limit and request:
+

> {prod-short} does not set CPU limits and requests by default. But it is still possible to configure CPU limits

Reviewer: ```suggestion
{prod-short} does not set CPU limits and requests by default. However, it is possible to configure CPU limits
```

> for `chePlugin` and `cheEditor` types via `meta.yaml` or devfile in the same way as it done for memory limits.

Reviewer: ```suggestion
for the `chePlugin` and `cheEditor` types in the `meta.yaml` file or in the devfile in the same way as it done for memory limits.
```

+
.The `chePlugin` `che-incubator/typescript/latest`
====
.`meta.yaml` spec section:
[source,yaml]
----
spec:
 containers:
   - image: docker.io/eclipse/che-remote-plugin-node:next
     name: vscode-typescript
     cpuLimit: 2000m
     cpuRequest: 500m
----

It results in a container with the following CPU limit and request:

|===
|CPU limit | 2 cores
|CPU request | 0.5 cores
|===
====

> To set CPU limits and request globally, there is dedicated environment variables in {prod-short}.

Reviewer: ```suggestion
To set CPU limits and requests globally, use the following dedicated environment variables:
```

> For `chePlugin` and `cheEditor` types, those are `+CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__LIMIT__CORES+`

Reviewer: ```suggestion
|===
| `chePlugin` | `+CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__LIMIT__CORES+`
| `cheEditor` | `+CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__REQUEST__CORES+`
|===

See also link:https://www.eclipse.org/che/docs/che-7/advanced-configuration-options-for-the-che-server-component/[Advanced configuration options for the {prod-short} server component].
```

> and `+CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__REQUEST__CORES+`. Please refer to variables description page for more information.

Reviewer: ```suggestion
```

> Also, need to keep in mind that the `LimitRange` object of Kubernetes namespace may specify some defaults for CPU limits and requests set by cluster administrators, so setting limits on application or workspace levels must comply with those settings,

Reviewer: ```suggestion
Note that the `LimitRange` object of the Kubernetes namespace may specify defaults for CPU limits and requests set by cluster administrators. To prevent start errors due to resources overrun, limits on application or workspace levels must comply with those settings.
```

> otherwise it is possible to get workspace startup errors due to resources overrun.

Reviewer: ```suggestion
```

.. Custom requirements for `dockerimage` components
+

> If present, the `memoryLimit` and `memoryRequest` attribute of the devfile defines the memory limit of a `dockerimage` container. {prod-short} sets automatically the memory request to match the memory limit in case it is not specified explicitly.

Reviewer: ```suggestion
If present, the `memoryLimit` and `memoryRequest` attributes of the devfile define the memory limit of a `dockerimage` container. {prod-short} automatically sets the memory request to match the memory limit in case it is not specified explicitly.
```
